### PR TITLE
Remove targets* platform condition from project references

### PR DESCRIPTION
**Pull request by Anipik**
Related to https://github.com/dotnet/runtime/issues/46338

### CODE REVIEW

File: src/libraries/System.Net.Http/src/System.Net.Http.csproj [hunk 1]
```diff
@@ -674,4 +673,7 @@
   <ItemGroup>
     <None Include="Resources\SR.resx" />
   </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Security.Cryptography.OpenSsl\src\System.Security.Cryptography.OpenSsl.csproj" />
```
Comment (safern) [inline src/libraries/System.Net.Http/src/System.Net.Http.csproj:677]: OpenSSL is a Unix only project, can this cause any issues on non-Unix?

Comment (Anipik) [reply src/libraries/System.Net.Http/src/System.Net.Http.csproj:677]: We have a platform independent configuration in openssl

File: src/libraries/System.Drawing.Common/src/System.Drawing.Common.csproj [hunk 2]
```diff
@@ -354,7 +354,7 @@
       <LogicalName>placeholder.ico</LogicalName>
     </EmbeddedResource>
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
+  <ItemGroup Condition="$(TargetFramework.StartsWith('netcoreapp3.0')) or $(TargetFramework.StartsWith('$(NetCoreAppCurrent)'))">
```
Comment (ViktorHofer) [inline src/libraries/System.Drawing.Common/src/System.Drawing.Common.csproj:357]: Why do you need this condition?

File: src/libraries/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj [hunk 3]
```diff
@@ -35,7 +35,7 @@
   <ItemGroup>
     <Reference Include="System.Runtime" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
+  <ItemGroup Condition="$(TargetFramework.StartsWith('$(NetCoreAppCurrent)'))">
```
Comment (ViktorHofer) [inline src/libraries/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj:38]: This project only targets `NetCoreAppCurrent`, why is this change required?

**Comment by ViktorHofer** [comment]
I'm worried about the additional references added to the assemblies. Did you consider setting Targets* properties to true during restore? A project is being restored when `<MSBuildRestoreSessionId />` is not empty.

**Comment by Anipik** [comment]
> I'm worried about the additional references added to the assemblies

I think the compiler removes the additional references to assemblies which are not used. I verified this as well using system.drawing.common dll, where the system.events references is only present in windows assembly.

**Review by safern — APPROVED** [review]
So long you have made sure the produced packages/assemblies are the same as before the change.

**Comment by ViktorHofer** [comment]
That's right, but it also increases build times on configurations that don't need the P2P. Did you consider setting the Target* to true during restore?

**Comment by Anipik** [comment]
> That's right, but it also increases build times on configurations that don't need the P2P. Did you consider setting the Target* to true during restore?

I can take a look at setting Targets* true during restore
